contrib/database/sql/internal/dsn.go

```diff
@@ -15,6 +15,9 @@ import (
 // ParseDSN parses various supported DSN types into a map of key/value pairs which can be used as valid tags.
 func ParseDSN(driverName, dsn string) (meta map[string]string, err error) {
 	meta = make(map[string]string)
+	if driverName != "mysql" && driverName != "postgres" && driverName != "sqlserver" && strings.Contains(dsn, ":") {
+		driverName = dsn[:strings.Index(dsn, ":")]
+	}
```

Comment:
I'm curious why you needed to do this, did you have an error while running the tests?

Reply:
Yes, the "driverName" for the tests is always "test" so the DSN is never parsed, even though it is valid.  I thought this was better than putting in "test" to the switch statement.

I'm going to add this into the default section with a recursive call and comment